### PR TITLE
build: fix i18n script usage with nx

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Before
image

After:
image

## The issue
We had a wrong config for building the i18n.

- I deleted a duplicated file (lingui configure was twice)
- I made it extract literals from all libs and cowswap
- I added in the main package json a `i18n` script (plus forced it to be run after installing the dependencies as it used to be)
- Migrated i18n to NX
- Added the caching in NX so we can cache the extraction and compilation of literal (not working!)

## Help needed
A follow-up can be made to this PR to fix the caching issue.
As you see, I tried to specify the inputs and outputs to NX so it can be smart about caching

Obviously, I did something wrong there because is never using the CACHE

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]

File: libs/common-utils/src/environments.test.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ describe('Detect environments using host and path', () => {
     })
   })
 
-  describe('Is production', () => {
+  describe.only('Is production', () => {
```
Comment (alfetopito) [inline libs/common-utils/src/environments.test.ts:23]: Ops, should revert this

File: apps/cowswap-frontend/project.json [hunk 2]
```diff
@@ -71,6 +72,28 @@
...
+    },
+    "i18n:compile": {
+      "executor": "nx:run-commands",
+      "dependsOn": ["i18n:extract"],
+      "options": {
+        "command": "npx lingui compile"
+      },
+      "inputs": ["{projectRoot}/src/locales/*.po"],
+      "outputs": ["{projectRoot}/src/locales/*.js"]
+    },
```
Comment (alfetopito) [inline apps/cowswap-frontend/project.json:89]: Why not use a single command since `i18n` is a noop?

Comment (anxolin) [reply apps/cowswap-frontend/project.json:89]: good point. I guess I wanted to be explicit and replicate what we had. i18n will be calling for both extract and compile. I will delete one to reduce the number of items there

Comment (anxolin) [reply apps/cowswap-frontend/project.json:89]: 1a810df0a9b0f13b22a024b28c72a709cd21ba65

File: libs/common-utils/src/environments.test.ts (hunk 1 above)
Comment (anxolin) [reply libs/common-utils/src/environments.test.ts:23]: https://github.com/cowprotocol/cowswap/pull/3309

**Pull request merged**